### PR TITLE
Helm channel

### PR DESCRIPTION
**Pull request by swopebe**

### CODE REVIEW

**Review by rokej — APPROVED** [review]
Thank you

File: applications/reconcile_options.adoc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+= Adding reconcile options
+
+You can use the `apps.open-cluster-management.io/reconcile-option` annotation in individual resources to override the subscription-level reconcile option. 
+
+For example, if you add `apps.open-cluster-management.io/reconcile-option: replace` annotation in the subscription and add `apps.open-cluster-management.io/reconcile-option: merge` annotation in a resource YAML in the subscribed Git repository, the resource is merged on the target cluster while other resources are replaced.
+
+[#reconcile-frequency-git]
+== Reconcile frequency Git channel
+
+You can now select reconcile frequency options: `high`, `medium`, `low`, and `off` in channel configuration to avoid unnecessary resource reconciliations and therefore prevent overload on subscription operator.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: I would remove "now", it seems like we would say "You can now.." in the What's new page

File: applications/reconcile_options.adoc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+*Required access:* Administrator and cluster administrator
+
+See the following definitions of the `settings:attribute:<value>`:
+
+- `Off`: The deployed resources are not automatically reconciled. A change in the subscription custom resource triggers a reconciliation. You can add or update a label or annotation.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: Not required at all, but I would change subscription custom resource to `Subscription` custom resource

I would do this bc it's the name of the resource. Another suggestion is removing the words, custom resource and write, "A change in the subscription triggers a reconciliation". 

You also don't have to change it at all lol

File: applications/reconcile_options.adoc [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+- `Low`: The deployed resources are automatically reconciled every hour, even if there is no change in the source Git repository.
+- `Medium`: This is the default setting. The subscription operator compares the currently deployed commit ID to the latest commit ID of the source repository every 3 minutes and apply changes to target clusters when there is change. Every 15 minutes, all resources are re-applied from the source Git repository to the target clusters, even if there is no change in the repository.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: Add a comma after "3 minutes"

I would also change re-applied to reapplied

File: applications/reconcile_options.adoc [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+- `High`: The deployed resources are automatically reconciled every two minutes, even if there is no change in the source Git repository.
+
+You can set this using `apps.open-cluster-management.io/reconcile-rate` annotation in the channel custom resource that is referenced by subscription.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: ...using _the_ `apps.open-...`

File: applications/reconcile_options.adoc [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  annotations:
+    apps.open-cluster-management.io/git-path: <application1>
+    apps.open-cluster-management.io/git-branch: <branch1>
+spec:
+  channel: sample/git-channel
+  placement:
+    local: true
+----
+
+In the previous example, all subscriptions that use `sample/git-channel` get `low` reconciliation frequency.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: I suggest:

In the previous example, all subscriptions that use `sample/git-channel` have a `low` reconciliation frequency.

or 

In the previous example, all subscriptions that use `sample/git-channel` are assigned a `low` reconciliation frequency.

File: applications/reconcile_options.adoc [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    apps.open-cluster-management.io/git-branch: <branch1>
+spec:
+  channel: sample/git-channel
+  placement:
+    local: true
+----
+
+In the previous example, all subscriptions that use `sample/git-channel` get `low` reconciliation frequency. 
+
+Regardless of the `reconcile-rate` setting in the channel, a subscription can turn the auto-reconciliation `off` by specifying `apps.open-cluster-management.io/reconcile-rate: off` annotation in the subscription CR.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: same monospace suggestion, which also can be rejected lol

File: applications/reconcile_options.adoc [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+----
+
+See that the resources deployed by `git-subscription` are never automatically reconciled even if the `reconcile-rate` is set to `high` in the channel.
+
+[#reconcile-frequency-helm]
+== Reconcile frequency Helm channel
+
+Every 15 minutes, the subscription operator compares currently deployed hash of your Helm chart to the hash from the source repository. Changes are applied to target clusters. The frequency of resource reconciliation impacts the performance of other application deployments and updates. 
+
+For example, if there are hundreds of application subscriptions and you to reconcile all subscriptions more frequently, the response time of reconciliation is slower.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: ..and you want to reconcile all subscriptions

File: applications/reconcile_options.adoc [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+Depending on the Kubernetes resources of the application, appropriate reconciliation frequency can improve performance.
+
+- `Off`: The deployed resources are not automatically reconciled. A change in the subscription CR triggers a reconciliation. You can add or update a label or annotation.
+- `Low`: The subscription operator compares currently deployed hash to the hash of the source repository every hour and apply changes to target clusters when there is change.
+- `Medium`: This is the default setting. The subscription operator compares currently deployed hash to the hash of the source repository every 15 minutes and apply changes to target clusters when there is change.
+- `High`: The subscription operator compares currently deployed hash to the hash of the source repository every 2 minutes and apply changes to target clusters when there is change.
+
+You can set this using `apps.open-cluster-management.io/reconcile-rate` annotation in the channel CR that is referenced by subscription. Here is an example.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: You can set this using _the...

Similar to my previous monospace comment, I would update channel CR to `Channel` CR. I would remove here is an example and only prompt the user with "See the following `helm-channel` example:

File: applications/reconcile_options.adoc [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    packageOverrides:
+    - path: spec
+      value:
+        defaultBackend:
+          replicaCount: 3
+  placement:
+    local: true
+----
+
+In this example, all subscriptions that uses `sample/helm-channel` get `low` reconciliation frequency.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: ...all subscriptions that use `sample/helm-channel` have a `low` reconciliation frequency.

File: applications/reconcile_options.adoc [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+    packageOverrides:
+    - path: spec
+      value:
+        defaultBackend:
+          replicaCount: 3
+  placement:
+    local: true
+----
+
+In this example, the resources deployed by `helm-subscription` will never be automatically reconciled even if the `reconcile-rate` is set to `high` in the channel.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: Future tense being used here, I suggest:

In this example, the resources deployed by `helm-subscription` are not automatically reconciled even if the `reconcile-rate` is set to `high` in the channel.


Also remember to add the path to this file into the main.adoc file

**Review by dockerymick — COMMENTED** [review]
Made some suggestions, let me know if you have questions

File: applications/reconcile_options.adoc (hunk 2 above)
Comment (swopebe) [reply applications/reconcile_options.adoc]: it's a good change, I am also changing **triggers** in this doc, I don't even think that is approved!

Comment (swopebe) [reply applications/reconcile_options.adoc]: triggers -- initiates

File: applications/reconcile_options.adoc (hunk 5 above)
Comment (swopebe) [reply applications/reconcile_options.adoc]: I like the second one, I didn't catch that but "get" is def not good.

File: applications/reconcile_options.adoc [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+[#reconcile-frequency-helm]
+== Reconcile frequency Helm channel
+
+Every 15 minutes, the subscription operator compares currently deployed hash of your Helm chart to the hash from the source repository. Changes are applied to target clusters. The frequency of resource reconciliation impacts the performance of other application deployments and updates. 
+
+For example, if there are hundreds of application subscriptions and you want to reconcile all subscriptions more frequently, the response time of reconciliation is slower.
+
+Depending on the Kubernetes resources of the application, appropriate reconciliation frequency can improve performance.
+
+- `Off`: The deployed resources are not automatically reconciled. A change in the subscription CR triggers a reconciliation. You can add or update a label or annotation.
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: Based on your comments, update triggers to initiates

File: applications/reconcile_options.adoc [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+      value:
+        defaultBackend:
+          replicaCount: 3
+  placement:
+    local: true
+----
+
+In this example, all subscriptions that uses `sample/helm-channel` are assigned a `low` reconciliation frequency. 
+
+Regardless of the reconcile-rate setting in the channel, a subscription can turn the auto-reconciliation `off` by specifying `apps.open-cluster-management.io/reconcile-rate: off` annotation in the subscription CR, as displayed in the following example:
```
Comment (dockerymick) [inline applications/reconcile_options.adoc]: monospace for subscription

**Review by dockerymick — APPROVED** [review]
Made some more comments, also add the file name to the main.adoc file. Otherwise, this looks good! Thanks for accepting my suggestions :D

File: applications/reconcile_options.adoc (hunk 11 above)
Comment (swopebe) [reply applications/reconcile_options.adoc]: oops, missed one

**Pull request merged**